src/conn/opts.rs

```diff
@@ -461,6 +461,84 @@ impl OptsBuilder {
         OptsBuilder { opts: opts.into() }
     }
 
+    /// Use a HashMap for creating creating an OptsBuilder instance:
+    /// ```ignore
+    /// OptsBuilder::new().from_has_map(client);
+    /// ```
+    /// `HashMap` key,value pairs:
+    /// - user = Username
+    /// - password = Password
+    /// - host = Host name or ip address
+    /// - port = Port, default is 3306
+    /// - socket = Unix socket or pipe name(on windows) defaults to `None`
+    /// - db_name = Database name (defaults to `None`).
+    /// - prefer_socket = Prefer socket connection (defaults to `true`)
+    /// - tcp_keepalive_time_ms = TCP keep alive time for mysql connection (defaults to `None`)
+    /// - compress = Compression level(defaults to `None`)
+    /// - tcp_connect_timeout_ms = Tcp connect timeout (defaults to `None`)
+    /// - stmt_cache_size = Number of prepared statements cached on the client side (per connection)
+    ///
+    /// Login .cnf file parsing lib https://github.com/rjcortese/myloginrs returns a HashMap for client configs
+    ///
+    /// **Note:** You do **not** have to use myloginrs lib.    
+    pub fn from_hash_map(mut self, client: &HashMap<String, String>) -> Result<Self, UrlError> {
+        for (key, value) in client.iter() {
+            match key.as_str() {
+                "user" => self.opts.0.user = Some(value.to_string()),
+                "password" => self.opts.0.pass = Some(value.to_string()),
+                "host" => {
+                    let host = url::Host::parse(value)
+                        .unwrap_or_else(|_| url::Host::Domain(value.to_owned()));
+                    self.opts.0.ip_or_hostname = host;
+                }
+                "port" => {
+                    let parsed: u16 = value.parse::<u16>().unwrap_or(3306);
```

Comment:
Could you please emit the `InvalidValue` for parse errors here and for other parameters? I'm trying to be as rigorous as possible here to avoid data loss, that is possible if errors like this are silently ignored.
The rule of thumb here is that if an option value is explicitly specified, then the client will either start with the specified value of an option or won't start at all

Reply:
@blackbeam 

Thanks for the feedback, I added `InvalidValue` to all parsing errors as suggested.
Also added:
- Another test case to assert for errors raised
- Added `PartialEq` to `Optsbuilder`, let me know if that is okay. If not, I can do a string check in the error case vs a type of error check.

Thanks again for feedback! It's my first time contributing to a project on git